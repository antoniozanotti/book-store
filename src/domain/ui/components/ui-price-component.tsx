import { UiButtonComponentInterface } from "./ui-price-component-interface";
import { Prisma } from "@prisma/client";
import { NumberFormatBase } from "react-number-format";

export function UiPriceComponent({ ...props }: UiButtonComponentInterface) {
  const value = new Prisma.Decimal(props.value ?? 0);
  props.value = value.toFixed(2);

  function format(value: string) {
    return new Intl.NumberFormat("en-US", {
      style: "currency",
      currency: "USD",
      maximumFractionDigits: 2,
    }).format(parseFloat(value));
  }

  return (
    <NumberFormatBase
      {...props}
      format={format}
      displayType="text"
      valueIsNumericString={true}
      renderText={(value) => <>{value}</>}
    />
  );
}
