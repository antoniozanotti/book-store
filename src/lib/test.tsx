import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
