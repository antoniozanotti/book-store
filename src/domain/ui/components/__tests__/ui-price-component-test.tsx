import { render, screen } from "@/lib/test";
import { UiPriceComponent } from "../ui-price-component";
import { productWithPagEandBookMock } from "@/domain/product/__tests__/product-with-page-and-book-mock";

describe("UiPriceComponent", () => {
  it("should render", async () => {
    const { container } = render(<UiPriceComponent value={productWithPagEandBookMock.price.toString()} />);
    expect(container).toMatchSnapshot();
  });
});