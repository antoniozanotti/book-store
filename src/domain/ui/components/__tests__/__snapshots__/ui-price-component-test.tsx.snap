// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiPriceComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    $302.00
  </div>
</div>
`;
